src/java.base/share/classes/sun/security/ssl/SSLSessionImpl.java

```diff
@@ -306,6 +316,14 @@ final class SSLSessionImpl extends ExtendedSSLSession {
      *       < length in bytes> PSK identity
      *   Anonymous
      *     < 1 byte >
+     * < 1 byte > exporterMasterSecret algorithm length (if == 0, no Key)
+     *   < length in bytes > exporterMasterSecret algorithm
+     *   < 2 bytes > exporterMasterSecretKey length
+     *   < length in bytes> exporterMasterSecretKey
+     * < 1 byte > Length of clientRandom
+     *   < length in bytes > clientRandom
+     * < 1 byte > Length of serverRandom
+     *   < length in bytes > serverRandom
      */
 
     SSLSessionImpl(HandshakeContext hc, ByteBuffer buf) throws IOException {
```

Comment:
this constructor is used for session resumption using deserialized stateless session tickets. AFAICT the resumed session uses a new set of exporter secrets (exporterMasterSecret is overwritten during processing of the Finished message, not sure about the randoms). Does it make any sense to store the original secrets in the stateless ticket?

Reply:
Good catch, I am 99% convinced neither is not needed.  I'll be removing these changes.   

For TLS 1.3, the exporter is recalculated at the end of the handshake for the new session.

For 1-1.2, the Session is indeed pulled from the serialized form, but the new session's Hello Random values are used with the existing MasterSecret to load the `TlsKeyMaterialParameterSpec` and obtain all of the keying material.

Reply:
Update, will push soon.

Had to move the random capture locations, as they need to capture the new client/server randoms instead of when the PMS is calculated.